lib/Wikia/src/Tasks/Tasks/BaseTask.php

```diff
@@ -427,6 +427,27 @@ public static function batch( array $tasks, $priority = null ) {
 		return AsyncTaskList::batch( $taskLists, $priority );
 	}
 
+	/**
+	 * Schedule this task to be published to the queue after the main response has been flushed back to the client.
+	 */
+	public function scheduleAsDeferredUpdate() {
+		$deferredUpdate = new class( $this ) implements \DeferrableUpdate {
+
+			/** @var BaseTask $task */
+			private $task;
+
+			public function __construct( BaseTask $task ) {
+				$this->task = $task;
+			}
+
+			function doUpdate() {
+				$this->task->queue();
+			}
+		};
```

Comment:
Can we actually make a class from this inline code to make this code easier to read?